### PR TITLE
update MSAL testcases with MSAL Sdk Wrapper

### PR DESCRIPTION
**Pull request by ayushmeghwani**
All MSAL Test Cases are updated with MSAL Sdk Wrapper

### CODE REVIEW

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                new IdLabB2cSisoPolicyPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        },TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        // ------ do silent request ------
+
+        IAccount account = getAccount();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java:110]: I don't think this will give you the account now that the wrappers are being used. I suggest you make the `getAccount` method in `MsalSdk` `public` and then use that here

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                .scopes(Arrays.asList(mScopes))
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authSilentResult = msalSdk.acquireTokenSilent(authTestSilentParams, TokenRequestTimeout.SILENT);
+        authSilentResult.assertSuccess();
+
+        // ------ do force refresh silent request ------
+
+        account = getAccount();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java:125]: Same comment as above

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase99563Clone.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        },TokenRequestTimeout.SHORT);
+
+        authResult.assertSuccess();
+
+        final IAccount account = getAccount();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase99563Clone.java:82]: This is not going to work. The tests are failing :(

As stated in another comment, since now the wrapper is used you would need to use the `getAccount` method in `MsalSdk` (you can make it `public`) - that should achieve the purpose here...if it does not then you can create a method in there that does what we are looking for here...but we can no longer use the `getAccount` method that is in `AbstractMsalUiTest`

I'll wait until you make these fixes and then review again as most tests cases will be broken right now because of this.

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Tests are failing as we calling the wrong `getAccount` method (see my comments).

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase99563Clone.java (hunk 3 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase99563Clone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java (hunk 2 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java (hunk 1 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/b2c/B2CIdLabSisoPolicyTestClone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+        final IAccount account = getAccount();
+
+        //acquiring token silently
+        final MsalAuthTestParams silentParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .authority(account.getAuthority())
+                .resource(mScopes[0])
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java:94]: The test is failing probably because this will not work without login hint

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(30));
+
+        // acquiring token silently after expiring AT
+        final MsalAuthTestParams refreshTokenParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .authority(account.getAuthority())
+                .resource(mScopes[0])
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java:110]: would need loginHint here too

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                        // cancel enroll here to short circuit as enroll will be started manually from CP anyway
+                        .enrollPageResponse(UiResponse.DECLINE)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java:95]: I don't think this needs to succeed given that we are declining enrollment in the UI....which should lead to cancel and hit on cancel callback...when then translates to MsalUserCancelException....and then for us that it is just an Exception so we should really be asserting failure here....

Are you observing something different?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+        // enroll device with CP
+        final IMdmAgent mdmAgent = (IMdmAgent) mBroker;
+        mdmAgent.enrollDevice(username, password);
+
+        // SECOND REQUEST WITH LOGIN HINT
+        final TokenRequestLatch latchTryAcquireAgain = new TokenRequestLatch(1);
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java:102]: I don't think the latch is needed anymore

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938365Clone.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                .activity(mActivity)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938365Clone.java:66]: would need to handle browser first run....see the original test case

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938368Clone.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+                final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(mLoginHint)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(true)
+                        .isFederated(true)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938368Clone.java:79]:
```suggestion
                new MicrosoftStsPromptHandler(promptHandlerParameters)
```

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938384Clone.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938384Clone.java:62]: would need to pass the authority....you can see the original test case for reference

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940393Clone.java [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    @Test
+    public void test_938447() throws Throwable {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(mLoginHint)
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940393Clone.java:58]: login hint should NOT be supplied here....see original test case for reference

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+                        .broker(mBroker)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .expectingLoginPageAccountPicker(false)
+                        .registerPageExpected(true)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java:96]: you probably want to do a LONG timeout here...at least that's what the original test case did

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+
+        authResult.assertSuccess();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java:98]: Just asserting success won't be enough here...see original test case for reference....in this test case we also need to assert that we got the `deviceid` claim back in the ID Token claims....previously this assertion was happening in the callback...now it needs to happen here (probably right after this line you can just add logic to assert for the claim)

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Quite a few test cases are broken....I left suggestions for how to fix them.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java (hunk 4 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java (hunk 5 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832430Clone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java (hunk 7 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java]: removed.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938365Clone.java (hunk 8 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938365Clone.java:69]: added the browser handler.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938368Clone.java (hunk 9 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938368Clone.java]: updated.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938384Clone.java (hunk 10 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/usgov/arlington/TestCase938384Clone.java]: passed.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java (hunk 12 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java]: added.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940393Clone.java (hunk 11 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940393Clone.java]: removed.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java (hunk 6 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java]: changed the assert function

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java (hunk 13 above)
Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java:105]: add the deviceid assert function

**Review by shahzaibj — COMMENTED** [review]
Overall looks good now, you can delete the original ones and rename these to remove clone

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase850455.java [hunk 14]
```diff
@@ -1,140 +0,0 @@
...
-import com.microsoft.identity.internal.testutils.labutils.LabConstants;
-import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
-
-import org.junit.Test;
-
-import java.util.concurrent.CountDownLatch;
-
-// [MSAL] Broker Auth for non-joined account - login
-// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/850455
-public class TestCase850455 extends AbstractMsalBrokerTest {
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase850455.java:50]: Why is this file deleted?

Comment (ayushmeghwani) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase850455.java:50]: I removed it mistakenly, added back and verified it.

**Review by shahzaibj — APPROVED** [review]
